### DISCUSSION TITLE
Support for the newer Gradle "plugins {}" block

### DISCUSSION THREAD

**Discussion by Mario-paul** [dipien/semantic-version-gradle-plugin#43] [General]
So I am a relatively newbie developer, and due to an issue with Gradle Plugin version in Android Studio Chipmunk, I was unable to use this plugin. So I upgraded to AS Dolphin, and created a new test project to learn how to use this plugin. When trying to apply the plugin, I noticed the root build.gradle looks completely different than that of the readme in this repo or of my older projects made in previous AS versions. 

```
// Top-level build file where you can add configuration options common to all sub-projects/modules.
buildscript {
    repositories {
        google()
        mavenCentral()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:7.4.0-rc03'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.7.21"

        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}
```

```
// Top-level build file where you can add configuration options common to all sub-projects/modules.
plugins {
    id 'com.android.application' version '7.4.0-rc03' apply false
    id 'com.android.library' version '7.4.0-rc03' apply false
    id 'org.jetbrains.kotlin.android' version '1.7.21' apply false
}
```

It seems that the newer AS releases now use the "newer" Gradle `plugins {}` block in project build.gradle, instead of the legacy `buildScript{}` block. After much research and fiddling around, I was able to make it work. But it doesn't look like [anything in the official Gradle Documentation](https://docs.gradle.org/7.6/userguide/plugins.html).

```
// Top-level build file where you can add configuration options common to all sub-projects/modules.
plugins {
    id 'com.android.application' version '7.4.0-rc03' apply false
    id 'com.android.library' version '7.4.0-rc03' apply false
    id 'org.jetbrains.kotlin.android' version '1.7.21' apply false
    id 'com.dipien.android.semantic-version' version '1.4.1' apply false
}

version = "1.0.0"

apply plugin: "com.dipien.android.semantic-version"
```

Nothing else I tried seemed to work. I am using both the `plugins{}` block as well as the old `apply plugin: ""` statement/method, which seems a bit hacky. Is it possible you can support the plugins{} block with an update? Or at the very least, document how to workaround like I did in the Readme since I lost a ton of hours researching this issue. I'm sure a lot of new users of this plugin would benefit greatly.

Sources:
https://stackoverflow.com/questions/74534360/my-build-gradle-project-level-looks-different-android-studio-dolphin
https://stackoverflow.com/questions/71247686/explain-new-format-of-top-level-gradle-file
https://stackoverflow.com/questions/32352816/what-the-difference-in-applying-gradle-plugin
https://stefma.medium.com/its-time-to-ditch-the-buildscript-block-a1ab12e0d9ce

**Comment by maxirosson** [comment]
The problem with the new way to apply plugins is that you can't define the project version before applying the plugin, unless you do what you did: apply false and then manually apply it. The plugin needs a version defined before the apply to work. 
So, basically you have two options: your approach or the readme approach (you can even mix the new and the legacy apply way in the same project)

**Reply by Mario-paul** [reply]
I see, thanks for the confirmation Maxi. I have a few more questions but they are unrelated so I will make another discussion. 👍
